internal/resources/fabric/precision_time/resource.go

```diff
@@ -0,0 +1,463 @@
+package precision_time
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"slices"
+	"time"
+
+	"github.com/equinix/equinix-sdk-go/services/fabricv4"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: framework.NewBaseResource(
+			framework.BaseResourceConfig{
+				Name: "equinix_fabric_precision_time",
+			},
+		),
+	}
+}
+
+type Resource struct {
+	framework.BaseResource
+}
+
+func (r *Resource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = resourceSchema(ctx)
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+
+	var plan PrecisionTimeModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Retrieve the API client from the provider metadata
+	client := r.Meta.NewFabricClientForFramework(ctx, req.ProviderMeta)
+
+	createRequest, diags := buildCreateRequest(ctx, plan)
+
+	resp.Diagnostics.Append(diags...)
+	if diags.HasError() {
+		return
+	}
+
+	ept, _, err := client.PrecisionTimeApi.CreateTimeServices(ctx).PrecisionTimeServiceRequest(createRequest).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Precision Time Service",
+			equinix_errors.FormatFabricError(err).Error(),
+		)
+		return
+	}
+
+	createTimeout, diags := plan.Timeouts.Create(ctx, 10*time.Minute)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	createWaiter := getCreateUpdateWaiter(ctx, client, ept.GetUuid(), createTimeout)
+	eptChecked, err := createWaiter.WaitForStateContext(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to create Precision Time Service %s", ept.GetUuid()), err.Error())
+		return
+	}
+
+	// Parse API response into the Terraform state
+	resp.Diagnostics.Append(plan.parse(ctx, eptChecked.(*fabricv4.PrecisionTimeServiceCreateResponse))...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```

Comment:
Is this where you're running into issues due to fields that are only present in the request and not in the response?

Reply:
Yea, this is the spot. It's related to advance_configuration only at this point. I'm considering separating ntp and ptp into their own attributes: 

`advance_configuration_ntp` and `advance_configuration_ptp`. 

NTP will be in request only and not in response. So it could be marked optional to allow for it being set early and not set later. 

PTP will be in request and response so it could stay as optional/computed. 

The nested object structure in terraform-plugin-framework along with it's strict checking of attribute behavior has proven difficult for me. Hoping this uncovers an appropriate path forward that doesn't actually involve modifying the schemas to be so different from the API schema similar to how `project_id` was handled. It can backfire if the `project` object gets extended or if `advance_configuration` gets extended.

Reply:
This also happens during Read and Update. Read when importing a resource, and Update when the response doesn't contain the details it expects. Though I haven't been able to test those as much because the Create has gotten a plan error every time `advance_configuration` is used. When `advance_configuration` is not used then we're able to proceed without plan errors based on parsing method putting some zero values there.

Reply:
It looks like the data source and resource have the same attributes.  Is the data source able to load the `advance_configuration` details on read?  Or does that attribute not work at all for data sources?

Is there some other data in the API response that corresponds to the `advance_configuration` settings?  Or a different API endpoint that can be used fetch the advanced configuration details?